javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/instrumentation/indy/AdviceTransformer.java

```diff
@@ -67,6 +67,9 @@ static byte[] transform(byte[] bytes) {
             }));
 
     TransformationContext context = new TransformationContext();
+    if (justDelegateAdvice) {
+      context.disableReturnTypeChange();
+    }
```

Comment:
[for reviewer] without this the advice return type is assumed to be an array, which means `skipOnIndex` is added when using `skipOn` annotation attribute, and this breaks when advice is not inlined.